contrib/parseq-zk-client/src/main/java/com/linkedin/parseq/zk/client/PlanLocal.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2016 LinkedIn, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.linkedin.parseq.zk.client;
+
+import com.linkedin.parseq.Task;
+import com.linkedin.parseq.promise.Promises;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+
+/**
+ * A {@link Map} that is shared across {@link Task}s belong to the same plan.
+ *
+ * @author Ang Xu
+ */
+public final class PlanLocal {
```

Comment:
Because the operations need to be deferred until after the plan has been submitted and started. Doing it at task composition time defects the purpose of this class.